### PR TITLE
Test for LENS / SOAP / tSOAP

### PR DESCRIPTION
**Pull request by SimoneMartino98**
Related Issues: 
Requested Reviewers: @andrewtarzia 
*Note for Reviewers: If you accept the review request add a :+1: to this post*

### CODE REVIEW

**Comment by SimoneMartino98** [comment]
Final parameters:
SOAP
`atol=1e-2, rtol=1e-2`
tSOAP
`atol=1e-11, rtol=1e-11`

The rtol for tSOAP can be smaller than E-11 but we already have a very low tolerance so i think that will not be a problem for possible code modifications

File: tests/test_lens.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+from pathlib import Path
```
Comment (andrewtarzia) [inline tests/test_lens.py:1]: tests should be in their own directory. So put the lens test in /tests/lens, and the soap tests in tests/soap/

File: tests/test_lens.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+import dynsight
+import h5py
+import numpy as np
+
+"""
```
Comment (andrewtarzia) [inline tests/test_lens.py:7]: Move this to the doc string of the tests.

File: tests/test_lens.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+Dynsyght function tested: dynsight.lens.list_neighbours_along_trajectory()
+                          dynsight.lens.neighbour_change_in_time()
+
+r_cuts checked: 2.5, 3.0, 3.5, 4.0, 4.5, 5.0, 5.5
+"""
+
+
+def test_lens_signals() -> None:
+    # i/o files
```
Comment (andrewtarzia) [inline tests/test_lens.py:22]: Comments should be full sentences.

File: tests/test_lens.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    input_file = "tests/systems/2_particles.hdf5"
```
Comment (andrewtarzia) [inline tests/test_lens.py:23]: use the `pathlib.Path` library for all paths

File: tests/test_lens.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            )
+
+        # check if control and test array are equal
+        assert np.array_equal(check_lens_nn, test_lens_nn), (
+            f"LENS analyses provided different values "
+            f"compared to the control system "
+            f"for r_cut: {lens_cutoffs[i]} (results: {output_file})."
+        )
+        # if test passed remove test_lens_nn array from test folder
+        Path(output_file).unlink()
```
Comment (andrewtarzia) [inline tests/test_lens.py:66]: Make it a path above and then use `output_file.unlink()`

File: tests/test_soap.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
+from pathlib import Path
+
+import dynsight
+import h5py
+import numpy as np
+
+"""
```
Comment (andrewtarzia) [inline tests/test_soap.py:7]: Move to test doc string

File: tests/test_soap.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+                            --> soaplmax = 8
+                            --> soapnmax = 8
+
+r_cuts checked: 1.75, 2.0, 2.15, 2.3, 2.45, 2.60, 2.75
+"""
+
+
+def test_soap_vectors() -> None:
+    # i/o files
+    input_file = "tests/systems/octahedron.hdf5"
```
Comment (andrewtarzia) [inline tests/test_soap.py:23]: As above - use pathlib.Path. 

All comments should be full sentences.

File: tests/test_soap.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+                    values as a control calculation at different r_cut.
+
+Control file path: tests/systems/octahedron.hdf5
+
+Dynsyght function tested: dynsight.soapify.saponify_trajectory()
+                            --> soaplmax = 8
+                            --> soapnmax = 8
+
+r_cuts checked: 1.75, 2.0, 2.15, 2.3, 2.45, 2.60, 2.75
+"""
```
Comment (andrewtarzia) [inline tests/test_soap.py:18]: Add explanation of machine dependance.

**Review by andrewtarzia — CHANGES_REQUESTED** [review]
Minor formatting changes required.

File: tests/test_lens.py (hunk 2 above)
Comment (SimoneMartino98) [reply tests/test_lens.py]: i think i don't understand what you mean

Comment (andrewtarzia) [reply tests/test_lens.py]: Move the comments from line 7 to after line 21 (make them to doc string to `test_lens_signals`). Do this for both test files.

File: tests/test_lens.py (hunk 3 above)
Comment (andrewtarzia) [reply tests/test_lens.py]: Full sentences include grammar (so, `# Define input and output files.`)

**Review by andrewtarzia — CHANGES_REQUESTED** [review]
Still many changes incomplete. But the move to specific folders is good.

**Comment by SimoneMartino98** [comment]
All required changes should have been done.

**Comment by andrewtarzia** [comment]
@SimoneMartino98  please do not resolve conversations in PRs. It is up to the reviewer to do so after they deem it complete.

File: tests/lens/test_lens.py [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+    Dynsyght function tested:
+        - dynsight.lens.list_neighbours_along_trajectory()
+        - dynsight.lens.neighbour_change_in_time()
+
+    r_cuts checked:
+        - [2.5, 3.0, 3.5, 4.0, 4.5, 5.0, 5.5]
+    """
+    # Define input and output files
+    original_dir = Path.cwd()
```
[the remark this answers is not in the document]
Comment (andrewtarzia) [reply tests/lens/test_lens.py:25]: Do this for each script.

**Review by andrewtarzia — CHANGES_REQUESTED** [review]
Almost done!

File: tests/lens/test_lens.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (SimoneMartino98) [reply tests/lens/test_lens.py]: Done

[the remark this answers is not in the document]
Comment (SimoneMartino98) [reply tests/lens/test_lens.py]: Is that ok?

`    # Define input and output files
    original_dir = Path(__file__).absolute().parent
    input_file = original_dir / "../systems/octahedron.hdf5"
    output_file = original_dir / "../octahedron_test_tsoap.hdf5"
`

**Review by andrewtarzia — APPROVED** [review]
Done!

**Pull request merged**